addons/web/static/src/js/core/owl_popover.js

```diff
@@ -0,0 +1,234 @@
+            // be in the dom when the component is instanciated
+            if (typeof this.props.target === 'string') {
+                this.target = document.querySelector(this.props.target);
+            } else {
+                this.target = this.props.target;
+            }
+            if (!this._isInViewport(this.target)) {
+                // target is no longer in the viewport, close the popover
+                return this._close();
+            }
+            this.requestedPosition = this.props.position;
+
+            const positionIndex = this.orderedPositions.indexOf(
+                this.requestedPosition
+            );
+
+            const positioningData = this.constructor._computePositioningData(
+                this.popoverRef.el,
+                this.target
+            );
+
+            // check if the requested position fits the viewport; if not,
+            // try all other positions and find one that does
+            const position = this.orderedPositions
+                .slice(positionIndex)
+                .concat(this.orderedPositions.slice(0, positionIndex))
+                .map((pos) => positioningData[pos])
+                .find((pos) => {
+                    this.popoverRef.el.style.top = `${pos.top}px`;
+                    this.popoverRef.el.style.left = `${pos.left}px`;
+                    return this._isInViewport(this.popoverRef.el);
+                });
+
+            // remove all positioning classes
+            this.orderedPositions.forEach((pos) => {
+                this.popoverRef.el.classList.remove(`o_popover--${pos}`);
+            });
+
+            /* apply the found position ('position' props or another position
+            that fits the viewport) if it fits the viewport, otherwise
+            fallback to 'bottom' if no position fits the viewport */
+            if (position) {
+                this.popoverRef.el.classList.add(`o_popover--${position.name}`);
+            } else {
+                this.popoverRef.el.style.top = `${positioningData.bottom.top}px`;
+                this.popoverRef.el.style.left = `${positioningData.bottom.left}px`;
+                this.popoverRef.el.classList.add(`o_popover--bottom`);
+            }
+        }
+
+        //--------------------------------------------------------------------------
+        // Private
+        //--------------------------------------------------------------------------
+
+        /**
+         * Send an event signaling that the popover must be closed.
+         * @private
+         */
+        _close() {
+            this.trigger('close-popover');
+        }
+
+        /**
+         * Check if the element is in the viewport.
+         * @param {HTMLElement}
+         * @returns {Boolean} True if the popover fits in the viewport, false otherwise.
+         */
+        _isInViewport(element) {
+            const rect = element.getBoundingClientRect();
```

Comment:
A little while ago I came across the problem of computing the size of an element.
I read those kinds of posts: https://stackoverflow.com/questions/10787782/full-height-of-a-html-element-div-including-border-padding-and-margin

summing up: getboundingrect doesn't account for margins. Though, I'm okay with the computation, if popver doesn't have margins